WordPress/openverse
# Use Elasticsearch configuration file to avoid tricky envvar names

## Fixes



Fixes https://github.com/WordPress/openverse/issues/4974 by @zackkrida 

## Description


See [my comment on the issue clarifying the cause and long-term fixes](https://github.com/WordPress/openverse/issues/4974#issuecomment-2375037131) for additional rationale of this change.

TL;DR: Docker compose broke the way environment variables are parsed, and no longer accepts those defined in `es/env.docker`. On the other hand, it appears that there is no way to pass allow-origin as an environment variable, as ES will not recognise the same option with an underscore, and provides no method to map hyphens. The configuration file is much more searchable against the docs anyway.

## Testing Instructions

CI must pass (proving ES is accessible throughout the development stack). Also check out this branch and run `ov j api/up` to get ES running locally, and make an http request like `ov http http://0.0.0.0:50292/_cluster/health`. Also use a tool like Elasticvue to prove CORS is working for local development usage.

## Checklist


- [N/A] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
